### ISSUE TITLE
Access to GL Extensions from dmsdk

### ISSUE THREAD

**Issue by dsmolyakov-ms** [defold/defold#6385]
For internal tasks, sometimes we need to access GL Extensions (the list of which in the engine is formed using glGetStringi(GL_EXTENSIONS, ...).
It would be very convenient to be able to get them by simply including the SDK and calling something like
dmGraphics::IsExtensionSupported(...)

Here, for example, you can see that this is implemented, but there is no access to it from the outside via the dmsdk, but I would like to: [graphics_opengl](https://github.com/defold/defold/blob/d393bae6a361f86cf2263ab312c9b3cea45253ab/engine/graphics/src/opengl/graphics_opengl.cpp)

For now, we had to duplicate the functionality already described in the engine and put it in external library.

**Comment by AGulev** [comment]
related https://github.com/defold/defold/issues/3350
https://github.com/defold/defold/issues/3790

**Comment by dsmolyakov-ms** [comment]
It is important to note that, for example, to correctly determine the support for some texture compression formats, such as ASTC and ETC2, it is also necessary to check the data from GL_COMPRESSED_TEXTURE_FORMATS.
[https://github.com/defold/defold/blob/d393bae6a361f86cf2263ab312c9b3cea45253ab/engine/graphics/src/opengl/graphics_opengl.cpp](graphics_opengl.cpp), line 1015:
`#define CASE(_NAME1,_NAME2) case _NAME1 : context->m_TextureFormatSupport |= 1 << _NAME2; break;`

That is, it turns out that you need access to the internal context->m_TextureFormatSupport or something like that.

**Comment by britzl** [comment]
Expand dmGraphics API to include

```
bool IsTextureFormatSupported(HContext context, TextureFormat format);
TextureFormat GetSupportedCompressionFormat(HContext context, TextureFormat format, uint32_t width, uint32_t height);
```

**Comment by Jhonnyg** [comment]
Closed by #6904
